pqarrow/builder/optbuilders.go

```diff
@@ -637,6 +638,22 @@ func (b *OptInt32Builder) Set(i int, v int32) {
 	b.data[i] = v
 }
 
+// UnsafeSwap swaps without bounds check.
+func (b *OptInt32Builder) UnsafeSwap(i, j int) {
+	b.data[i], b.data[j] = b.data[j], b.data[i]
+}
+
+// UnsafeSet sets v at index i without bounds check.
+func (b *OptInt32Builder) UnsafeSet(i int, v int32) {
+	b.data[i] = v
+	bitutil.SetBit(b.validityBitmap, i)
+}
+
+// UnsafeValue returns value at index i without bounds check.
+func (b *OptInt32Builder) UnsafeValue(i int) int32 {
+	return b.data[i]
```

Comment:
I don't think we need a bounds check in `Value` since this is performed implicitly by the runtime (just that the panic would be different. We can probably remove this `UnsafeValue` and remove the `if` from `Value`. What do you think? cc @metalmatze since I think you introduced this code.

Reply:
Not exactly sure anymore. Let's see how it goes now that it's merged. Happy to revise in the future.

Reply:
The problem is that we now have duplicate methods that do the same thing (and one version seems to have a bug with setting nulls IIRC from another comment on this PR). We should probably fix this sooner rather than later.

Reply:
@metalmatze @asubiotto I created a PR for this #680 we can move this discussion there.